tests/e2e/simple_rp_roa_test.py

```diff
@@ -228,30 +238,46 @@ def test_setup(self):
 
     @pytest.mark.parametrize("service", [Routinator, FortValidator, OctoRPKI, Rcynic, RPKIClient, RPKIValidator3])
     def test_rtr(self, docker_host_fqdn, docker_project, function_service_manager, service, metadata):
+        #
+        # Use Docker Compose to deploy the given Relying Party service and its dependendencies.
+        # On tear down the service container and its dependent containers will be killed and removed.
+        #
         function_service_manager.start_services_with_dependencies(docker_project, service.name)
 
-        try:
-            rtr_start_time = int(time())
-            logging.info(f'Connecting RTR client to {docker_host_fqdn}:{service.rtr_port}')
-            received_roas = set(rtr_fetch_one(docker_host_fqdn, service.rtr_port, service.rtr_timeout_seconds))
-            rtr_elapsed_time = int(time()) - rtr_start_time
-
-            # r is now a list of PFXRecord
-            # see: https://python-rtrlib.readthedocs.io/en/latest/api.html#rtrlib.records.PFXRecord
-            logging.info(f'Received {len(received_roas)} ROAs via RTR from {service.name} in {rtr_elapsed_time} seconds')
+        def retry_if_sync_timeout(exception):
+            return isinstance(exception, rtrlib.exceptions.SyncTimeout)
 
-            # are each of the TEST_ROAS items in r?
-            # i.e. is the intersection of the two sets equal to that of the TEST_ROAS set?
-
-            logging.info(f'Comparing {len(received_roas)} received ROAs to {len(TEST_ROAS)} expected ROAs...')
-            expected_roas = set([roa_to_roa_string(r) for r in TEST_ROAS])
-            assert received_roas == expected_roas
-        except rtrlib.exceptions.SyncTimeout as e:
-            logging.error(f'Timeout (>{service.rtr_timeout_seconds} seconds) while syncing RTR with {service.name} at {docker_host_fqdn}:{service.rtr_port}')
+        @retry(
+            stop_max_attempt_number=3,
```

Comment:
This 3 retry attempts setting was added for the rcynic -> custom script to produce JSON -> Lighthttpd case which for some reason librtr fails to connect to and to see that there are VRPs available if it connects too soon, but on a retry it works just fine. It seems it might be related to the way RTRTR produces an error in this situation that GoRTR does not but Martin said he believes that the RTRTR behaviour is correct, in combination with a timing difference with the other tests that also use the same "custom script to produce JSON -> Lighthttpd" pattern but don't exhibit a problem.

However, writing it like this causes the other RP tests to also be retried if they fail, but for those RPs that just delays the test result, especially in the case of the RIPE NCC Validator 3 as that is configured in the test framework (see the rpki-deploy repo) to wait for 240 seconds *per try*. With this in mind the test framework also defines the number of retries per RP and sets them all to 1 except for Rcynic which it sets to 3. However, that number isn't used here. I'm not sure if I can pass a variable to `stop_max_attempt_number` here.